everyvoice/demo/app.py

```diff
@@ -163,137 +172,224 @@ def normalize_text(text: str) -> str:
                 )
-    if speakers == ["all"]:
-        speak_list = model_speakers
-    else:
-        for speaker in speakers:
-            if speaker in model_speakers:
-                speak_list.append(speaker)
-            else:
-                print(
-                    f"Attention: The model have not been trained for speech synthesis with '{speaker}' speaker. The '{speaker}' speaker option will not be available for selection."
-                )
-
-    if lang_list == []:
+    if language_list == []:
         raise ValueError(
             f"Language option has been activated, but valid languages have not been provided. The model has been trained in {model_languages} languages. Please select either 'all' or at least some of them."
         )
-    if speak_list == []:
-        raise ValueError(
-            f"Speaker option has been activated, but valid speakers have not been provided. The model has been trained with {model_speakers} speakers. Please select either 'all' or at least some of them."
-        )
-    default_lang = lang_list[0]
-    interactive_lang = len(lang_list) > 1
-    default_speak = speak_list[0]
-    interactive_speak = len(speak_list) > 1
+
+    return language_list
+
+
+def load_app_ui_labels(
+    ui_config_json_path: os.PathLike | None = None,
+    speakers: list[str] = ["all"],
+    languages: list[str] = ["all"],
+    model_languages: list[str] = [],
+    model_speakers: list[str] = [],
+) -> tuple[GradioChoices, GradioChoices, dict[str, str]]:
+    """Load the app config JSON file if provided and update the speaker and language lists.
+    This function checks if the provided ui_config_json_path is a valid JSON file and contains
+    the 'speakers' and 'languages' keys. If they are present, it updates the speaker and language lists accordingly.
+    Args:
+        ui_config_json_path (os.PathLike | None): Path to the app config JSON file.
+        speakers (list[str]): List of speakers to be used in the app.
+        languages (list[str]): List of languages to be used in the app.
+        speak_list (list[str]): List of speakers from the model.
+        lang_list (list[str]): List of languages from the model.
+    Returns:
+        tuple: A tuple of [display-name, form-value] for speakers, languages and the app config json.
+    Raises:
+        ValueError: If the 'speakers' or 'languages' are not checkpoint or keys in the app config JSON do not match the provided speakers or languages, or if the keys are not dictionaries.
+    """
+    # tuple of (display-name, form-value) for gradio dropdown
+    language_list: GradioChoices = set_language_list(languages, model_languages)
+    speaker_list: GradioChoices = set_speaker_list(speakers, model_speakers)
+    app_ui_labels = {}  # dict[str,str]  # app UI config JSON
+
+    # json config file is passed
+    if ui_config_json_path and str(ui_config_json_path).lower().endswith(".json"):
```

Comment:
Marc's question is a good one: why do we insist on a .json file extension?

If we require it, then we should issue an error message when there is a ui_config_json_path and it does not end with .json.

Reply:
I removed this bit